exonum/src/api/node/public/explorer.rs

```diff
@@ -171,6 +171,46 @@ impl AsRef<[u8]> for TransactionHex {
     }
 }
 
+/// Status of a call.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CallStatusInfo {
+    /// Call status
+    pub status: Result<(), ExecutionError>,
+    /// Call execution proff
+    pub call_proof: MapProof<CallInBlock, ExecutionError>,
+}
+
+/// Call status response.
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct CallStatusResponse {
+    /// Status of transaction
+    pub transaction: Option<CallStatusInfo>,
+    /// Status of `before_transaction` hook
+    pub before_transactions: Option<CallStatusInfo>,
+    /// Status of `after_transaction` hook
+    pub after_transactions: Option<CallStatusInfo>,
+}
+
+/// The kind of requested call
+#[derive(Debug, Clone, Copy, Eq, Serialize, Deserialize, PartialEq, Hash)]
+pub enum CallKind {
+    /// Call kind of transaction
+    Transaction,
+    /// Call kind of `before_transactions` hook
+    BeforeTransactions,
+    /// Call kind of `after_transactions` hook
+    AfterTransactions,
+}
+
+/// Call status query parameters.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub struct CallStatusQuery {
```

Comment:
To be honest, it looks like an attempt to mix everything at once.
I'd expect the query to be something like:

```
pub enum CallKind {
    Transaction(hash),
    BeforeTransactions(height),
    AfterTransactions(height),
}
```

All three types of the request are different and I'm not sure that we should support requesting data for more than one at the time (meaning that if user is interested in the transaction result, before_transactions result and after_transactons result, he should do three different requests to the endpoint).

Reply:
Left as is because in case with height it need instance id to build CallInBlock

Reply:
And why can't we include `InstanceId` to the request?
```
pub enum CallKind {
    Transaction(hash),
    BeforeTransactions(height, instance_id),
    AfterTransactions(height, instance_id),
}
```

Requesting result of the service hook with hash of the transaction seems logically incorrect: what if there was no transactions for the service within block, and we still want to see the result of `AfterTransactions`?

And it's overall inconvenient and counterintuitive for users to request not a tx by tx hash.